app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def self.update_user_posts_counter(user)
+    user.update(post_counter: Post.where(author_id: user.id).length)
+  end
```

Comment:
- Please replace `user` in line 7 with `author` and remove the parameter `user` from the update_user_posts_counter method. Please keep in mind that the author has already been defined on line 2.
- A faster way to update the post_counter would be:
```ruby
def update_user_posts_counter
  author.increment!(:posts_counter)
end
```